docs/plans/go-ci/plan.cue

```diff
@@ -0,0 +1,24 @@
+package main
+
+import (
+	"dagger.io/dagger"
+
+	"universe.dagger.io/go"
+	//"universe.dagger.io/bash"
+)
+
+dagger.#Plan & {
+	client: filesystem: "./hello": read: contents: dagger.#FS
+
+	actions: {
+		test: go.#Test & {
+			source:  client.filesystem."./hello".read.contents
+			package: "./..."
+		}
+
+		build: go.#Build & {
+			source: client.filesystem."./hello".read.contents
+		}
```

Comment:
The first thing that comes to mind with this is multi-platform support. If dagger CLI runs on macOS (which is what the majority of users use) then Docker will be Linux, meaning that the resulting binary will be Linux by default. Maybe we should mention this somewhere.

Reply:
That's a very good point
We should extract the `go env GOOS` and `go env GOARCH` from the host to build for the current host platform.
I guess, this should go in the `go` pkg.

Reply:
A `TODO` will work for now.